<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ControlView</class>
 <widget class="QWidget" name="ControlView">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>353</width>
    <height>391</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Merge Sort Controls</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <layout class="QHBoxLayout" name="listLengthLayout">
     <item>
      <widget class="QLabel" name="listLengthLabel">
       <property name="text">
        <string>List length</string>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="listSizeSpinBox">
       <property name="value">
        <number>1</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="drawButton">
       <property name="text">
        <string>Draw</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QCheckBox" name="stepModeCheckBox">
       <property name="text">
        <string>Step mode</string>
       </property>
       <property name="tristate">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="continueButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="maximumSize">
        <size>
         <width>100</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="text">
        <string>Continue</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="waitDurationSpinBox">
       <property name="value">
        <number>1</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="waitDurationButton">
       <property name="maximumSize">
        <size>
         <width>50</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="text">
        <string>sec</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="sortButton">
       <property name="text">
        <string>Sort</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pauseButton">
       <property name="text">
        <string>Pause</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="resetButton">
       <property name="text">
        <string>Reset</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout">
     <property name="spacing">
      <number>6</number>
     </property>
     <property name="sizeConstraint">
      <enum>QLayout::SetDefaultConstraint</enum>
     </property>
     <item>
      <widget class="QLabel" name="initialListTitleLabel">
       <property name="text">
        <string>Initial list</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="initialListLineEdit">
       <property name="text">
        <string>1, 2, 3, 4, 5, 6, 7, 8, 9, 10</string>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Sorted list</string>
       </property>
       <property name="textFormat">
        <enum>Qt::AutoText</enum>
       </property>
       <property name="scaledContents">
        <bool>false</bool>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
       </property>
       <property name="wordWrap">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="sortedListLineEdit">
       <property name="text">
        <string>1, 2, 3, 4, 5, 6, 7, 8, 9, 10</string>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Status message:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPlainTextEdit" name="statusPlainTextEdit">
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>100</height>
        </size>
       </property>
       <property name="font">
        <font>
         <italic>true</italic>
        </font>
       </property>
       <property name="plainText">
        <string>. Ipso facto</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>sortButton</sender>
   <signal>clicked()</signal>
   <receiver>ControlView</receiver>
   <slot>_sortSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>112</x>
     <y>99</y>
    </hint>
    <hint type="destinationlabel">
     <x>137</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pauseButton</sender>
   <signal>clicked()</signal>
   <receiver>ControlView</receiver>
   <slot>_pauseSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>220</x>
     <y>99</y>
    </hint>
    <hint type="destinationlabel">
     <x>308</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>resetButton</sender>
   <signal>clicked()</signal>
   <receiver>ControlView</receiver>
   <slot>_resetSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>329</x>
     <y>99</y>
    </hint>
    <hint type="destinationlabel">
     <x>339</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>drawButton</sender>
   <signal>clicked()</signal>
   <receiver>ControlView</receiver>
   <slot>_drawSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>329</x>
     <y>36</y>
    </hint>
    <hint type="destinationlabel">
     <x>339</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stepModeCheckBox</sender>
   <signal>stateChanged(int)</signal>
   <receiver>ControlView</receiver>
   <slot>_stepModeSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>45</x>
     <y>65</y>
    </hint>
    <hint type="destinationlabel">
     <x>66</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>continueButton</sender>
   <signal>clicked()</signal>
   <receiver>ControlView</receiver>
   <slot>_continueSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>202</x>
     <y>71</y>
    </hint>
    <hint type="destinationlabel">
     <x>248</x>
     <y>-16</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>waitDurationButton</sender>
   <signal>clicked()</signal>
   <receiver>ControlView</receiver>
   <slot>_waitDurationSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>311</x>
     <y>57</y>
    </hint>
    <hint type="destinationlabel">
     <x>41</x>
     <y>-14</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>_sortSlot()</slot>
  <slot>_pauseSlot()</slot>
  <slot>_resetSlot()</slot>
  <slot>_drawSlot()</slot>
  <slot>_stepModeSlot()</slot>
  <slot>_continueSlot()</slot>
  <slot>_waitDurationSlot()</slot>
 </slots>
</ui>
